File: HW01/CPP/src/LinkedList.cpp
#include "../include/LinkedList.h"
#include <iostream>


LinkedList::LinkedList() {
    this->head = nullptr;
    this->count = 0;
}

Node* LinkedList::getHead() {
    return this->head;
}

void LinkedList::insertAtBeginning(Item* item, string type, int stock) {
    Node* newNode = new Node();
    this->count++;
    newNode->item = item;
    newNode->type = type;
    newNode->stock = stock;
    newNode->ID = this->count;
    newNode->next = this->head;
    this->head = newNode;
}

void LinkedList::insertAtEnd(Item* item, string type, int stock) {
    Node* newNode = new Node();
    this->count++;
    newNode->item = item;
    newNode->type = type;
    newNode->ID = this->count;
    newNode->stock = stock;
    newNode->next = nullptr;

    if (!this->head) {
        head = newNode;
        return;
    }

    Node* temp = this->head;
    while (temp->next) {
        temp = temp->next;
    }

    temp->next = newNode;
}

File: HW01/PYTHON/project/LinkedList.py
class Node:
    def __init__(self, item, type, stock):
        self.item = item
        self.type = type
        self.stock = stock
        self.next = None
        self.ID = None
        
    def setID(self, ID):
        self.ID = ID
        
    def setNext(self, next):
        self.next = next

class LinkedList:
    def __init__(self):
        self.head = None
        self.count = 0
        
    def getHead(self):
        return self.head
    
    
    def insertAtBeginning(self, item, type, stock):
        newNode = Node(item, type, stock)
        self.count += 1
        newNode.setID(self.count)
        newNode.setNext(self.head)
        self.head = newNode